### ISSUE TITLE
JVMTI VThreadMonitorTest fails with an unexpected monitor

### ISSUE THREAD

**Issue by dipak-bagadiya** [eclipse-openj9/openj9#16685]
### Issue
The JVMTI `VThreadMonitorTest` fails with an unexpected monitor from `GetCurrentContendedMonitor` for a virtual thread.

### Test CMD
```
make test TEST="jtreg:test/hotspot/jtreg/serviceability/jvmti/vthread/VThreadMonitorTest" JTREG="JAVA_OPTIONS=--enable-preview -Dvm.continuations=true;VERBOSE=all"
```

### Test Output:
```
checkContendedMonitor: VirtualThread-0: contended monitor: (nil)
FAIL: is_vt: 1: unexpected monitor from GetCurrentContendedMonitor
stack trace of current thread:
JVMTI Stack Trace for thread MainThread: frame count: 10
 0: VThreadMonitorTest: checkContendedMonitor(Ljava/lang/Thread;Ljava/lang/Object;Ljava/lang/Object;)V
 1: VThreadMonitorTest: main([Ljava/lang/String;)V
 2: java/lang/invoke/LambdaForm$DMH.0x000000004c014bf0: invokeStatic(Ljava/lang/Object;Ljava/lang/Object;)V
 3: java/lang/invoke/LambdaForm$MH.0x000000004c018a10: invoke(Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
 4: java/lang/invoke/LambdaForm$MH.0x000000004c019790: invokeExact_MT(Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Object;Ljava/lang/Object;)Ljava/lang/Object;
 5: jdk/internal/reflect/DirectMethodHandleAccessor: invokeImpl(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;
 6: jdk/internal/reflect/DirectMethodHandleAccessor: invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;
 7: java/lang/reflect/Method: invoke(Ljava/lang/Object;[Ljava/lang/Object;)Ljava/lang/Object;
 8: com/sun/javatest/regtest/agent/MainWrapper$MainThread: run()V
 9: java/lang/Thread: run()V


java.lang.RuntimeException: FAILED status returned from the agent
        at VThreadMonitorTest.main(VThreadMonitorTest.java:142)
        at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
        at java.base/java.lang.reflect.Method.invoke(Method.java:578)
        at com.sun.javatest.regtest.agent.MainWrapper$MainThread.run(MainWrapper.java:125)
        at java.base/java.lang.Thread.run(Thread.java:1573)
JavaTest Message: Test threw exception: java.lang.RuntimeException: FAILED status returned from the agent
JavaTest Message: shutting down test
```

**Comment by fengxue-IS** [comment]
@thallium can you take a look at this

**Comment by thallium** [comment]
In libVThreadMonitorTest.cpp:97 we are expecting the contended monitor to be either `lock0` or `lock2` but we are getting a null object.

**Comment by thallium** [comment]
The test also failed for platform thread. We should check the way of triggering contention to see if this is against the spec.

**Comment by thallium** [comment]
Currently in our code the JVMTI Monitor Contended Enter is triggered before setting the thread flag. When the JVMTI event is triggered, the contended monitor check is then triggered, but at that time we haven't set the flag so we couldn't get the correct monitor object. I've proposed a fix https://github.com/eclipse-openj9/openj9/pull/16844 to set the flag before triggering the event.

**Comment by babsingh** [comment]
@thallium Two more tasks before we can close this issue:
- The test needs to be re-enabled for JDK19 and JDK20; currently, it is disabled; and
- The fix will also need to be backported to the 0.37 and 0.38 release branches since it impacts all Java versions.

fyi @fengxue-IS @JasonFengJ9 @tajila

**Comment by thallium** [comment]
> @thallium Two more tasks before we can close this issue:
> 
> * The test needs to be re-enabled for JDK19 and JDK20; currently, it is disabled; and
> * The fix will also need to be backported to the 0.37 and 0.38 release branches since it impacts all Java versions.
> 
> fyi @fengxue-IS @JasonFengJ9 @tajila

Done, see https://github.com/eclipse-openj9/openj9/pull/16896 and https://github.com/adoptium/aqa-tests/pull/4420

**Comment by babsingh** [comment]
@thallium Thanks. Closing as per https://github.com/eclipse-openj9/openj9/issues/16685#issuecomment-1468627578.
